bot/src/bot.rs

```diff
@@ -100,12 +142,15 @@ impl Bot {
             None,
             None,
         )?;
+        let mut fetch_instances_count = 0;
 
         for app_info in saving_modules.modules {
             let code_id = app_info.module.reference.unwrap_app()?;
+
             let mut contract_addrs = daemon
                 .rt_handle
                 .block_on(utils::fetch_instances(daemon.channel(), code_id))?;
+            fetch_instances_count += contract_addrs.len();
```

Comment:
Count added before retain not sure if we want to include contracts without permission! You can set it to the length of `contract_instances_to_autocompound` after the loop if we don't want to include those

Reply:
This is for now just to get the tooling for metrics. Then we need to make a session where we ask what we want to track and adapt accordingly.